src/internal/easee/auth.go

```diff
@@ -61,15 +96,73 @@ func (a *authenticator) AccessToken() (string, error) {
 		return credentials.AccessToken, nil
 	}
 
+	if a.status == statusWaitingToReconnect || a.status == statusConnectionFailed {
+		return "", errors.New("connection interrupted, waiting for user to re-login")
+	}
+
 	newCredentials, err := a.http.RefreshToken(credentials.AccessToken, credentials.RefreshToken)
 	if err != nil {
-		return "", fmt.Errorf("failed to fetch new credentials: %w", err)
+		return "", a.handleFailedRefreshToken(err)
 	}
 
+	// reset backoff stats
+	a.status = statusWorkingProperly
+	a.attempts = 0
+
 	err = a.cfgSvc.SetCredentials(newCredentials.AccessToken, newCredentials.RefreshToken, newCredentials.ExpiresIn)
 	if err != nil {
 		return "", fmt.Errorf("failed to save credentials in storage: %w", err)
 	}
 
 	return newCredentials.AccessToken, nil
 }
+
+func (a *authenticator) Logout() error {
+	return a.cfgSvc.ClearCredentials()
+}
+
+// handleFailedRefreshToken sets a correct status when refresh operation has failed.
+// relies on mutex protection within a caller.
+func (a *authenticator) handleFailedRefreshToken(err error) error {
+	// we aren't able to refresh anymore. Requires user re-login
+	var httpError HTTPError
+	if ok := errors.As(err, &httpError); ok && httpError.Status == 401 {
+		_ = a.notificationManager.Event(&notification.Event{EventName: notificationEaseeStatusOffline})
```

Comment:
Same as my last comment: always evaluate errors.

Reply:
please just use `http.StatusUnauthorized` instead of plain `401`